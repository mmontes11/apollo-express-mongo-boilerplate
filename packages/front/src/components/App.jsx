import React from "react";
import Router from "components/Router";
import "styles/index.scss";

const App = () => (
  <div className="container">
    <Router />
  </div>
);

export default App;
