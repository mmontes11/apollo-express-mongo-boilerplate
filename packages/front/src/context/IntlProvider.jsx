import React from "react";
import PropTypes from "prop-types";
import { IntlProvider as ReactIntlProvider } from "react-intl";
import { getLocale, getMessages } from "config/localization";

const IntlProvider = ({ children }) => {
  const locale = getLocale();
  const messages = getMessages(locale);
  return (
    <ReactIntlProvider key={locale} locale={locale} messages={messages}>
      {children}
    </ReactIntlProvider>
  );
};

IntlProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default IntlProvider;
