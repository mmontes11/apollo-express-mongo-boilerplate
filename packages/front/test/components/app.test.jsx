import React from "react";
import { mount } from "enzyme";
import { ApolloProvider } from "@apollo/react-hooks";
import client from "apollo";
import App from "components/App";
import IntlProvider from "context/IntlProvider";

describe("components/app", () => {
  it("renders", () => {
    const wrapper = mount(
      <ApolloProvider client={client}>
        <IntlProvider>
          <App />
        </IntlProvider>
      </ApolloProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
