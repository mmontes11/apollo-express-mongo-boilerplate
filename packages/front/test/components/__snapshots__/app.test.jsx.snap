// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/app renders 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IntlProvider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <IntlProvider
      key="en"
      locale="en"
      messages={
        Object {
          "Book now!": "Book now!",
          "Contact us": "Contact us",
          "Error": "Error loading home cards",
          "How we work": "How we work",
          "More details": "More details",
          "The company": "The company",
          "front": "front",
        }
      }
    >
      <App>
        <div
          className="container"
        >
          <Connect(Messages)>
            <Messages
              getMessages={[Function]}
              sendMessage={[Function]}
            />
          </Connect(Messages)>
        </div>
      </App>
    </IntlProvider>
  </IntlProvider>
</Provider>
`;
